knimin/lib/data_access.py

```diff
@@ -2399,6 +2399,178 @@ def getProjectNames(self):
+        return self._con.execute_fetchone(sql, sql_args)[0]
+
+    def set_sample_plate_info(self, sample_plate_id, sample_plate_info):
+        """Sets attributes of a sample plate by ID
+
+        Parameters
+        ----------
+        sample_plate_id : int
+            ID of the sample plate to set attributes to
+        sample_plate_info: tuple of (str x2, datetime, str x2)
+            name, email, creation_timestamp, notes, plate_type
+
+        Returns
+        -------
+        bool
+            True if successful
+        """
+        sql = """UPDATE pm.sample_plate
+                 SET name = %s, email = %s, creation_timestamp = %s,
+                    notes = %s, plate_type_id = (SELECT plate_type_id
+                    FROM pm.plate_type WHERE name = %s)
+                 WHERE sample_plate_id = %s"""
+        sql_args = list(sample_plate_info + (sample_plate_id,))
+        for i in range(1, 4):
+            sql_args[i] = sql_args[i] or None
+        self._con.execute(sql, sql_args)
+        return True
+
+    def get_sample_plate_info(self, sample_plate_id):
+        """Gets attributes of a sample plate by ID
+
+        Parameters
+        ----------
+        sample_plate_id : int
+            ID of the sample plate to get attributes from
+
+        Returns
+        -------
+        tuple of (str x2, datetime, str x2)
+            name, email, creation_timestamp, notes, plate_type
+        """
+        sql = """SELECT p.name, p.email, p.creation_timestamp, p.notes,
+                    pm.plate_type.name
+                 FROM pm.sample_plate p
+                 JOIN pm.plate_type USING (plate_type_id)
+                 WHERE sample_plate_id = %s"""
+        return tuple(self._con.execute_fetchone(sql, [sample_plate_id]))
+
+    def delete_sample_plate(self, sample_plate_ids):
+        """Deletes sample plates and corresponding plate-to-sample maps
+
+        Parameters
+        ----------
+        sample_plate_ids : list of int
+            Sample plate IDs to delete
+
+        Returns
+        -------
+        bool
+            True if successful
+        """
+        sql = """DELETE FROM pm.sample_plate_layout
+                 WHERE sample_plate_id = %s"""
+        self._con.executemany(sql, [[x] for x in sample_plate_ids])
+        sql = """DELETE FROM pm.sample_plate
+                 WHERE sample_plate_id = %s"""
+        self._con.executemany(sql, [[x] for x in sample_plate_ids])
+        return True
+
+    def set_sample_plate_layout(self, sample_plate_id, sample_plate_layout):
+        """Sets sample plate layout by ID
+
+        Parameters
+        ----------
+        sample_plate_id : int
+            Sample plate ID
+        sample_plate_layout : list of tuple of (str, int, int, str, str)
+            Sample ID, column ID, row ID, name, notes
+
+        Returns
+        -------
+        bool
+            True if successful
+        """
+        sql = """DELETE FROM pm.sample_plate_layout
```

Comment:
same here w.r.t. transaction. basically, i worry about worst case scenario, either a disrupted connection, power outage, or exception between queries which could lead to state inconsistencies in the database through a partial execution of one of these methods.